functionary/ui/views/environments/detail.py

```diff
@@ -9,15 +9,7 @@
 
 class EnvironmentDetailView(LoginRequiredMixin, UserPassesTestMixin, DetailView):
```

Comment:
It's actually the opposite.  The `test_func` should have stayed and has now been restored.  The new generic views are built for use with models that have an environment.  The views for the Environment and Team models have to continue doing their own thing because neither one of those has an associated environment.

I made an attempt at making that distinction a little clearly in the `ui.views.generic` module docstring by adding this verbage:

>These are intended for use on model based views where the model is either directly or indirectly 
associated with an environment. That is, the model has a foreign key to `Environment`, or has a 
foreign key to another model that itself has a foreign key to `Environment`.

I suspect that the distinction will also cause people some slight confusion.  But the `Environment` and `Team` views should hopefully be the only real outliers in this.

Reply:
Ah ok, that makes more sense to me